pkg/cosign/tuf/client.go

```diff
@@ -683,9 +684,9 @@ func noCache() bool {
 	return b
 }
 
-func remoteFromMirror(ctx context.Context, mirror string) (client.RemoteStore, error) {
+func remoteFromMirror(mirror string) (client.RemoteStore, error) {
 	if _, parseErr := url.ParseRequestURI(mirror); parseErr != nil {
-		return GcsRemoteStore(ctx, mirror, nil, nil)
+		mirror = fmt.Sprintf("https://%s.storage.googleapis.com", mirror)
```

Comment:
For the CLI flag, I assume we'll continue to assume that the remote is a bucket if there's no scheme, and otherwise, HTTP?

Reply:
As long as the docs are consistent, I'm fine. I'm totally fine with removing all GCS references though. :)

Reply:
I think differentiating is nice, because it's simpler to just specify the bucket name, but if we want to converge, then we'll need to note that the full path is always required (which is a breaking client change, not sure if that's worth it)

Reply:
The default variable change won't cause problems.

I just remembered one thing:
The only place this change may cause problems is if someone is caching the remote location as a GCS bucket, e.g. `sigstore-tuf-root`, and then tries to call initialize with that. Your change as is will handle that since `remoteForMirror` will resolve URL parse errors as the GCS bucket. Just FYI. But otherwise good to go!

We can remove the parse error handling as a breaking change. It will mean clients will need to clear the cache, that's all. The error they get will also tell them to do that.

Reply:
I don't think we should force users to clear the cache. Is it significant to continue supporting both?

Reply:
I'm fine keeping the fallback for compatibility ~indefinitely really, just so long as we can drop the client dependency.